packages/paima-sdk/paima-db/migrations/up.sql

```diff
@@ -28,6 +28,12 @@ CREATE TABLE cde_tracking (
   done BOOLEAN NOT NULL DEFAULT false
 );
 
+CREATE TABLE cde_tracking_cardano (
+  slot INTEGER PRIMARY KEY,
+  datum_count INTEGER NOT NULL,
```

Comment:
Why do we need to store the datum count? When would we only be part-way through tracking a slot?

Reply:
is this any different than the evm events? There can be more than one tx per block, so there can be more than one delegation for example, and I guess something could go wrong in between? I mostly just replicated the way it works with the existing cde's, since it didn't look any different to me

Reply:
Oh you're right - this was dead code that I didn't remove. I fixed this now: https://github.com/PaimaStudios/paima-engine/pull/257